File: python/백준/집합과 맵/10815 숫자카드.py
n = int(input())

cards = list(map(int, input().split(" ")))

m = int(input())

checks = list(map(int, input().split()))

_dict = {}

for card in cards:
    _dict[card] = 0  # 아무 값이나 매핑한 거임

for check in checks:
    if check not in _dict:
        print(0, end=" ")
    else:
        print(1, end=" ")

File: cpp/백준/집합과 맵/10815 숫자카드.cpp
#include <iostream>
#include <unordered_set>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;
    unordered_set<int> _dict;
    for (int i = 0; i < n; ++i) {
        int card;
        cin >> card;
        _dict.insert(card);
    }

    int m;
    cin >> m;
    for (int i = 0; i < m; ++i) {
        int check;
        cin >> check;
        if (_dict.find(check) == _dict.end()) cout << 0 << " ";
        else cout << 1 << " ";
    }
    cout << "\n";
    return 0;
}
